core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -623,27 +623,31 @@ public void resetState() {
   public Map<String, Boolean> updateConfiguration(Map<String, String> propertiesMap) {
     Map<String, Boolean> result = new HashMap<>();
     int successCount = 0;
+    if (!Configuration.getBoolean(PropertyKey.CONF_DYNAMIC_UPDATE_ENABLED)) {
+      LOG.warn("Dynamic configuration update disabled but attempted, set {} to true to enable it",
+          PropertyKey.CONF_DYNAMIC_UPDATE_ENABLED.getName());
+      return result;
+    }
     for (Map.Entry<String, String> entry : propertiesMap.entrySet()) {
       try {
         PropertyKey key = PropertyKey.fromString(entry.getKey());
-        if (Configuration.getBoolean(PropertyKey.CONF_DYNAMIC_UPDATE_ENABLED)
-            && key.isDynamic()) {
+        if (key.isDynamic()) {
           Object oldValue = Configuration.get(key);
           Configuration.set(key, entry.getValue(), Source.RUNTIME);
           result.put(entry.getKey(), true);
           successCount++;
           LOG.info("Property {} has been updated to \"{}\" from \"{}\"",
               key.getName(), entry.getValue(), oldValue);
         } else {
-          LOG.debug("Update a non-dynamic property {} is not allowed", key.getName());
+          LOG.warn("Update a non-dynamic property {} is not allowed", key.getName());
           result.put(entry.getKey(), false);
         }
       } catch (Exception e) {
         result.put(entry.getKey(), false);
         LOG.error("Failed to update property {} to {}", entry.getKey(), entry.getValue(), e);
       }
     }
-    LOG.debug("Update {} properties, succeed {}.", propertiesMap.size(), successCount);
+    LOG.info("Update {} properties, succeed {}.", propertiesMap.size(), successCount);
```

Comment:
I think it makes more sense if you report the failed ones and report what the properties are exactly. So you need to keep a `failedSet = new HashSet<String>()` or just filter `result` with `entry.value==false`.